### DISCUSSION TITLE
How to implement a Select + Combobox Grid

### DISCUSSION THREAD

**Discussion by cloud-walker** [ariakit/ariakit#3946] [Questions] [answered]
I'm trying to achieve a similar result of the [Select Grid](https://ariakit.org/examples/select-grid) example, but I want it in a [Select with Combobox](https://ariakit.org/examples/select-combobox) scenario.

@diegohaz u have some advice on that? Here my attempt: https://stackblitz.com/edit/vitejs-vite-gk7snc?file=src%2FMyGridSelectCombobox.tsx

**Comment by diegohaz** [comment, marked as answer]
Here's a working version: https://stackblitz.com/edit/vitejs-vite-vtwyrv?file=src%2FMyGridSelectCombobox.tsx

It uses `ComboboxRow` and reverses the order of `ComboboxItem` and `SelectItem`. However, I agree this should be more intuitive. Could you please open an issue?

**Reply by cloud-walker** [reply]
Sure thing!

Btw my colleague @marilari88, mentioned a solution using the prop `rowId` (which I wasn't aware of): here the modified stackblitz: https://stackblitz.com/edit/vitejs-vite-adrmxc?file=src%2FMyGridSelectCombobox.tsx

**Reply by diegohaz** [reply]
Yes, `rowId` will handle keyboard interaction, but you still need to manually set the correct ARIA roles (`grid` -> `row` -> `gridcell`).

**Reply by cloud-walker** [reply]
another problem is responsive, if you need to have variable columns based on the available width, you need to dynamically create the rows, using some sort of JS based media queries (like a useMedia hook I guess)

**Reply by diegohaz** [reply]
Something like this will do the trick: https://codesandbox.io/p/sandbox/ariakit-grid-with-dynamic-size-8w27jv?file=%2Fsrc%2FApp.js

**Reply by cloud-walker** [reply]
your example is not behaving exactly like mine (aside from the working keyboard interaction), inverting SelectItem with ComboboxItem cause the aria-selected to be true on both actually selected items and active items.

Maybe this is correct accessibility wise, but from UI perspective that way we don't have a way to distinguish between active and selected items 🤔

**Reply by diegohaz** [reply]
You're right, it should only be on the selected item, not the active/focused one. In the future, `ComboboxItem` will stop adding `aria-selected` to the focused item (which remains for legacy reasons).

**Reply by cloud-walker** [reply]
ok, so it is a bug at the moment, is there a workaround for that? Maybe using the function version of the `render` prop?
